core/src/main/java/com/datastax/oss/driver/api/core/config/ProgrammaticConfigBuilder.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core.config;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * A builder that allows the creation of a config loader where options are overridden
+ * programmatically.
+ *
+ * @see DriverConfigLoader#programmaticBuilder()
+ */
+public interface ProgrammaticConfigBuilder extends OngoingConfigOptions<ProgrammaticConfigBuilder> {
```

Comment:
I think that both options of loading config: 
`Programmatic` and `from-file` should extend an interface that has a name stating that it is a builder.
Right now if the user will want to use `ProgrammaticConfigBuilder` he will need to use this class a variable type.
When he will want to switch to `from file loader` he will need to use `DriverExecutionProfile` as a variable type.
He can use: `OngoingConfigOptions<T>` but the name OngoingConfigOptions doesn't say that this a builder.

Reply:
I'm not following. The setters on `DriverExecutionProfile` are used to create new profiles on an already loaded configuration (see [derived profiles](https://docs.datastax.com/en/developer/java-driver/4.0/manual/core/configuration/#derived-profiles) in the manual), that's not specific to loading from a file or not. And they don't follow a builder pattern.